### PR TITLE
Homework5: Pavel Klepcha

### PR DESCRIPTION
**Pull request by klep4a**

### CODE REVIEW

File: src/Homework5/decor_result.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from datetime import datetime
+from runner_funcs import nod_evklid
+from runner_funcs import runner
+
+
+def decor_result(func):
+    def wr(*args, **kwargs):
+        with open('results.txt', 'a') as f:
+            f.write('Я функция: {} '
```
Comment (belovmd) [inline src/Homework5/decor_result.py]: it has troubles with encoding (you can check your file in pr)

File: src/Homework5/decor_result.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                    'Result: {}  {}\n'.format(func.__name__,
+                                              func(*args, **kwargs),
+                                              str(datetime.today())[:-10]))
```
Comment (belovmd) [inline src/Homework5/decor_result.py:17]: great

File: src/Homework5/runner_funcs.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          format(fizz_buzz.__name__, fizz_buzz.__doc__))
+    for i in range(1, 32):
+        print('Fizz' * (i % 3 == 0) + "Buzz" * (i % 5 == 0) or i, end=' ')
+
+
+def runner(*funcs):
+    all_funcs = {k: v for k, v in globals().items()
+                 if not k.startswith(('__', 'runner'))}
+    if funcs:
+        for i in [*funcs]:
```
Comment (belovmd) [inline src/Homework5/runner_funcs.py]: for i in funcs

File: src/Homework5/runner_funcs.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    """
+
+    print('Привет! Я функция {}, я делаю {}'.
+          format(fizz_buzz.__name__, fizz_buzz.__doc__))
+    for i in range(1, 32):
+        print('Fizz' * (i % 3 == 0) + "Buzz" * (i % 5 == 0) or i, end=' ')
+
+
+def runner(*funcs):
+    all_funcs = {k: v for k, v in globals().items()
```
Comment (belovmd) [inline src/Homework5/runner_funcs.py:80]: you should also check if it's callable

File: src/Homework5/get_ranges.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ get_ranges([4,7,10]) // "4,7,10"
+ get_ranges([2, 3, 8, 9]) // "2-3,8-9"
+"""
+
+
+def get_ranges(lst):
+    temp_list = []
+    ranges_list = []
+    for i in lst:
+        if i + 1 not in lst or i - 1 not in lst:
```
Comment (belovmd) [inline src/Homework5/get_ranges.py]: O(n**2) solution it can be solved for O(n) complexity

File: src/Homework5/imdb.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+import os
+
+
+def read_imdb():
+    try:
+        with open(os.path.join('data5', 'ratings.list')) as file:
+            line_count, source_dict = 1, {}
+            for line in file:
+                if 29 <= line_count < 279:
```
Comment (belovmd) [inline src/Homework5/imdb.py]: magic numbers can be moved to constants

File: src/Homework5/max_div_exp_of_two.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
+"""Вводится число найти его максимальный делитель,
+являющийся степенью двойки. 10(2) 16(16), 12(4).
+"""
+
+
+def div_exp_bit_length(num):
+    div = num.bit_length() - 1
+    while num % (1 << div):
```
Comment (belovmd) [inline src/Homework5/max_div_exp_of_two.py:8]: it can be checked with & it seems

**Review by belovmd — CHANGES_REQUESTED** [review]
- Please address next issues

File: src/Homework5/decor_result.py (hunk 2 above)
Comment (klep4a) [reply src/Homework5/decor_result.py:17]: Thank you

File: src/Homework5/max_div_exp_of_two.py (hunk 7 above)
Comment (klep4a) [reply src/Homework5/max_div_exp_of_two.py:8]: I`m not so familiar with these operators, unfortunately, but i will grow!

File: src/Homework5/runner_funcs.py (hunk 4 above)
Comment (klep4a) [reply src/Homework5/runner_funcs.py:80]: checked

File: src/Homework5/decor_result.py (hunk 1 above)
Comment (klep4a) [reply src/Homework5/decor_result.py]: It was win-1251 encoding in file, no idea, how it happens ;-), switch to utf-8

File: src/Homework5/runner_funcs.py (hunk 3 above)
Comment (klep4a) [reply src/Homework5/runner_funcs.py]: really, thank`s!

File: src/Homework5/get_ranges.py (hunk 5 above)
Comment (klep4a) [reply src/Homework5/get_ranges.py]: rewrited

File: src/Homework5/imdb.py (hunk 6 above)
Comment (klep4a) [reply src/Homework5/imdb.py]: rewrited

File: src/Homework5/max_div_exp_of_two.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+"""Вводится число найти его максимальный делитель,
```
Comment (belovmd) [inline src/Homework5/max_div_exp_of_two.py:1]: def binary(num):
    my_two = 2
    while 1:
        if num & my_two == 0:
            my_two = my_two << 1
        else:
            return my_two

**Review by belovmd — COMMENTED** [review]
8,5

**Comment by belovmd** [comment]
9

**Pull request closed without merging**